bibliopixel/layout/multimap.py

```diff
@@ -1,10 +1,33 @@
-class MultiMapBuilder(object):
+from . geometry import gen_matrix
 
-    def __init__(self):
+
+class MultiMapBuilder:
+
+    def __init__(self, make_object):
         self.map = []
         self.offset = 0
+        self.make_object = make_object
+
+    def make_drivers(self, driver, drivers):
+        if not drivers:
+            return [self.make_object(**driver)]
+
+        if driver:
+            # driver is a default for each driver.
+            drivers = [dict(driver, **d) for d in drivers]
+
+        return [self._make_driver(**d) for d in drivers]
+
+    def _make_driver(self, width, height, matrix=None, **kwds):
+        row = gen_matrix(width, height, **(matrix or {}))
+        for y, row_entry in enumerate(row):
+            self.map.append([i + self.offset for i in row_entry])
+
+        self.offset += len(row) * len(row[0])
+        return self.make_object(width=width, height=height, **kwds)
 
     def addRow(self, *maps):
+        # DEPRECATED
```

Comment:
`add_one_row` completely replaces `addRow`.

In practice `len(maps)` is always 1. Nearly all that code is involved with keeping track of the cases where `len(maps) > 1`.

(More, calling `add_one_row` repeatedly has exactly the same effect that `addRow` would have had with `maps` of length greater than 1.  !)